src/hooks/useSHakkaBalance.ts

```diff
@@ -0,0 +1,72 @@
+import { Contract as MulticallContract, Provider as MulticallProvider } from '@pelith/ethers-multicall';
+import { useWeb3React } from '@web3-react/core';
+import { ChainDataFetchingState, NEW_SHAKKA_ADDRESSES } from '../constants';
+import debounce from 'lodash.debounce';
+import { BigNumber } from '@ethersproject/bignumber';
+import { ChainId } from '../constants';
+import { useEffect, useMemo, useState } from 'react';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import { useBlockNumber } from '../state/application/hooks';
+import STAKING_ABI from '../constants/abis/shakka.json';
+
+export type SHakkaBalanceType = {
+    [chainId in ChainId]: BigNumber; 
+  };
+
+export default function useSHakkaBalance(): {
+  sHakkaBalanceInfo: SHakkaBalanceType;
+  fetchDataState: ChainDataFetchingState;
+  } {
+    const { account } = useWeb3React();
+    const latestBlockNumber = useBlockNumber();
+    const [sHakkaBalanceInfo, setSHakkaBalanceInfo] = useState<SHakkaBalanceType>();
+    const [transactionSuccess, setTransactionSuccess] = useState(false);
+  
+    const fetchDataState: ChainDataFetchingState = useMemo(() => {
+      return transactionSuccess ? ChainDataFetchingState.SUCCESS : ChainDataFetchingState.LOADING;
+    }, [transactionSuccess]);
+  
+    const ethProvider = new JsonRpcProvider(process.env.REACT_APP_NETWORK_URL);
+    const bscProvider = new JsonRpcProvider(process.env.REACT_APP_BSC_NETWORK_URL);
+    const polygonProvider = new JsonRpcProvider(process.env.REACT_APP_POLYGON_NETWORK_URL);
+    const kovanProvider = new JsonRpcProvider(process.env.REACT_APP_KOVAN_NETWORK_URL);
```

Comment:
after deep in source code, there has promise asynchronous process on contructor, maybe should add these to useRef / useMemo, prevent unhandled error?

Reply:
I think it is a good idea, cause this code will new the provider every rerender